src/python/dxpy/asset_builder.py

```diff
@@ -79,22 +82,30 @@ def validate_conf(asset_conf):
     """
     if 'name' not in asset_conf:
         raise AssetBuilderException('The asset configuration does not contain the required field "name".')
-    # TODO: this default is not a good idea, and we will have to remove it once we ask customers to always provide release
-    if 'release' not in asset_conf:
-        asset_conf['release'] = "12.04"
-    elif asset_conf['release'] not in ['16.04', '14.04', '12.04']:
-        raise AssetBuilderException('The "release" field value should be either "12.04" (DEPRECATED), "14.04", "16.04".')
+
+    # Validate runSpec
+    if 'release' not in asset_conf or asset_conf['release'] not in ['16.04', '14.04', '12.04']:
```

Comment:
That removes the ability to build assets for 12.04. Do we still allow building apps for 12.04? I would add to the title of this commit and to the CHANGELOG: "Remove ability to build assets for Ubuntu 12.04" or similar so that users know which older dxpy version to use if they really need to build 12.04 asset, e.g. to rebuild from source and repro old clinical trial results.

Reply:
It removes defaulting to 12.04, or defaulting to any release. It's still a valid release to build an asset.

Reply:
Ah, right.